include/libpmemobj++/transaction.hpp

```diff
@@ -446,6 +446,42 @@ class transaction {
 		transaction::run(pool, tx, locks...);
 	}
 
+	/**
+	 * Takes a “snapshot” of given elements of type T number (1 by default),
+	 * located at the given address ptr in the virtual memory space and
+	 * saves it to the undo log. The application is then free to directly
+	 * modify the object in that memory range. In case of a failure or
+	 * abort, all the changes within this range will be rolled back. The
+	 * supplied block of memory has to be within the pool registered in the
+	 * transaction. This function must be called during TX_STAGE_WORK. This
```

Comment:
TX_STAGE_WORK is a C-layer detail. In C++, you can just say that it must be called during/inside of a transaction.